cypress/support/fragments/data_import/dataImportViewAllPage.js

```diff
@@ -15,8 +26,239 @@ export default {
+      expect(dates).to.deep.equal(sortedDates);
+    });
+  },
+
+  checkByErrorsInImport({ filter }) {
+    this.waitUIToBeFiltered();
+    this.getMultiColumnListCellsValues(2).then(statuses => {
+      // if filter is 'Yes', then check for error otherwise, completed status
+      const expectedStatuses = filter === 'Yes' ? ['Failed', 'Completed with errors'] : ['Completed'];
+      // each status in the statuses array should be in the expectedStatuses array
+      const isFilteredByErrorStatus = statuses.every(jobStatus => expectedStatuses.includes(jobStatus));
+
+      // eslint-disable-next-line no-unused-expressions
+      expect(isFilteredByErrorStatus).to.be.true;
+
+      this.resetAllFilters();
+    });
+  },
+
+  checkByDate({ from, end }) {
+    const queryString = UrlParams.getDateQueryString({ from, end });
+    this.getNumberOfMatchedJobs(queryString).then(count => {
+      // ensure MultiColumnList is filtered by Date
+      this.checkRowsCount(count);
+
+      this.resetAllFilters();
+    });
+  },
+
+  checkByJobProfileName({ jobProfileName }) {
+    this.waitUIToBeFiltered();
+    this.getMultiColumnListCellsValues(4).then(jobProfiles => {
+      // each profile name in the jobProfiles array should be equal to jobProfileName
+      const isFilteredByProfile = jobProfiles.every(name => name === jobProfileName);
+
+      // eslint-disable-next-line no-unused-expressions
+      expect(isFilteredByProfile).to.be.true;
+
+      this.resetAllFilters();
+    });
+  },
+
+  checkByUserName({ userName }) {
+    this.waitUIToBeFiltered();
+    this.getMultiColumnListCellsValues(6).then(userNames => {
+      // each name in the userNames array should be equal to the userName
+      const isFilteredByUser = userNames.every(name => name === userName);
+
+      // eslint-disable-next-line no-unused-expressions
+      expect(isFilteredByUser).to.be.true;
+
+      this.resetAllFilters();
+    });
+  },
+
+  checkByInventorySingleRecord({ filter }) {
+    this.waitUIToBeFiltered();
+    cy.get('body').then($body => {
+      if ($body.find('#list-data-import').length) {
+        this.getMultiColumnListCellsValues(4).then(profiles => {
+          const inventorySingleRecordProfiles = [
+            'Inventory Single Record - Default Create Instance',
+            'Inventory Single Record - Default Update Instance'
+          ];
+
+          if (filter === 'Yes') {
+            const isInventorySingleRecord = profiles.every(profile => inventorySingleRecordProfiles.includes(profile));
+            // eslint-disable-next-line no-unused-expressions
+            expect(isInventorySingleRecord).to.be.true;
+          } else {
+            const isNotInventorySingleRecord = profiles.every(profile => !inventorySingleRecordProfiles.includes(profile));
+            // eslint-disable-next-line no-unused-expressions
+            expect(isNotInventorySingleRecord).to.be.true;
+          }
+        });
+      } else {
```

Comment:
do we have option to create records for such cases to exclude way with empty table after filtering?

Reply:
Unfortunately, not. I tried to create job profile with Inventory Single Record. It was unsuccessfull. So, I had to check List to be in the dom before getting cells. Can you recommend some way?

Reply:
@TetianaParanich could you help us?